simulation/traffic_simulator/src/data_type/entity_status.cpp

```diff
@@ -19,56 +19,78 @@ namespace traffic_simulator
-: entity_status_(canonicalize(may_non_canonicalized_entity_status, hdmap_utils, route_lanelets))
+: canonicalized_lanelet_pose_{may_non_canonicalized_entity_status.lanelet_pose_valid?std::optional(CanonicalizedLaneletPose(
+    may_non_canonicalized_entity_status.lanelet_pose, route_lanelets, hdmap_utils )):std::nullopt},
+      entity_status_{may_non_canonicalized_entity_status}
 {
+  canonicalize();
 }
 
 CanonicalizedEntityStatus::CanonicalizedEntityStatus(const CanonicalizedEntityStatus & obj)
-: entity_status_(static_cast<EntityStatus>(obj))
+: canonicalized_lanelet_pose_(obj.canonicalized_lanelet_pose_),
+  entity_status_(static_cast<EntityStatus>(obj))
 {
 }
 
-auto CanonicalizedEntityStatus::canonicalize(
-  const EntityStatus & may_non_canonicalized_entity_status,
-  const std::shared_ptr<hdmap_utils::HdMapUtils> & hdmap_utils) -> EntityStatus
+CanonicalizedEntityStatus & CanonicalizedEntityStatus::operator=(
+  const CanonicalizedEntityStatus & obj)
 {
-  auto canonicalized = may_non_canonicalized_entity_status;
-  if (may_non_canonicalized_entity_status.lanelet_pose_valid) {
-    canonicalized.lanelet_pose_valid = true;
-    canonicalized.lanelet_pose = static_cast<LaneletPose>(
-      CanonicalizedLaneletPose(may_non_canonicalized_entity_status.lanelet_pose, hdmap_utils));
-  } else {
-    canonicalized.lanelet_pose_valid = false;
-    canonicalized.lanelet_pose = LaneletPose();
-  }
-  return canonicalized;
+  this->canonicalized_lanelet_pose_ = obj.canonicalized_lanelet_pose_;
+  this->entity_status_ = obj.entity_status_;
+  return *this;
 }
 
-auto CanonicalizedEntityStatus::canonicalize(
-  const EntityStatus & may_non_canonicalized_entity_status,
-  const std::shared_ptr<hdmap_utils::HdMapUtils> & hdmap_utils, const lanelet::Ids & route_lanelets)
-  -> EntityStatus
+auto CanonicalizedEntityStatus::canonicalize() -> void
 {
-  auto canonicalized = may_non_canonicalized_entity_status;
-  if (may_non_canonicalized_entity_status.lanelet_pose_valid) {
-    canonicalized.lanelet_pose_valid = true;
-    canonicalized.lanelet_pose = static_cast<LaneletPose>(CanonicalizedLaneletPose(
-      may_non_canonicalized_entity_status.lanelet_pose, hdmap_utils, route_lanelets));
+  assert(entity_status_.lanelet_pose_valid == canonicalized_lanelet_pose_.has_value());
+  if (canonicalized_lanelet_pose_) {
+    entity_status_.lanelet_pose_valid = true;
+    entity_status_.lanelet_pose = static_cast<LaneletPose>(canonicalized_lanelet_pose_.value());
+    /*
+      The Oz position and orientation based on LaneletPose are rewritten to the used map_pose,
+      since such adjustment relative to the lanelet is necessary,
+      The position in Ox and Oy is not rewritten because the map_pose retrieved via
+      lanelet_pose = toCanonicalizedLaneletPose(map_pose), then map_pose = toMapPose(lanelet_pose)
+      can be different from the original one (especially if the entity changes lane)
```

Comment:
What is `Ox` ,`Oy`, `Oz` ?

Reply:
It is a little clearer, but the definition of Ox/Oy/Oz is unclear.
Please put “X/Y/Z values of the Cartesian coordinate system with respect to map” in the comment instead of omitting them.

When writing comments, please try as much as possible to write comments that can be interpreted correctly even without prior knowledge of the meaning.